pkg/protocol/http1/ext/stream.go

```diff
@@ -136,7 +128,7 @@ func (rs *bodyStream) Read(p []byte) (int, error) {
 				return 0, err
 			}
 			if chunkSize == 0 {
-				err = utils.SkipCRLF(rs.reader)
+				err = ReadTrailer(rs.trailer, rs.reader)
```

Comment:
It's a little bit strange that couple trailer parsing and streaming body‘s `Read` API here.

Reply:
What about lazy parse trailer, and raise an error if chunks are not parsed completed, when user calls the Trailer-related APIs?